lib/HttpException.php

```diff
@@ -24,8 +24,6 @@ interface HttpException
      *
      * This may either be just the number, or a number and a human-readable
      * message, separated by a space.
-     *
-     * @return string|null
      */
-    public function getHttpStatus();
+    public function getHttpStatus(): ?string;
```

Comment:
The PHP doc had `string|null` as the return type. I preserved that in the explicit declaration of return type `?string`

That has flow-on effects - this is saying that the integer `200` is not acceptable to return, it must return the string "200".

Implementations of this, like in `ClientHttpException` above, already had the return value declared as `int`

What to do? What is the "requirement"? Do we refactor to make all this allow both `string|int` to be passed around, or require `string` or require `int` ?